warehouse/query-core/src/main/java/datawave/query/scheduler/SequentialScheduler.java

```diff
@@ -129,9 +134,25 @@ public boolean hasNext() {
                     if (null != qd.getRanges())
                         rangesSeen += qd.getRanges().size();
                     count.incrementAndGet();
-                    if (null == newQueryData)
+                    if (null == newQueryData) {
                         newQueryData = new QueryData(qd);
-                    else {
+                        String transformedQuery = JexlStringBuildingVisitor.buildQuery(config.getQueryTree());
```

Comment:
Why are we doing this?  The QueryData object should have the query that is appropriate to run for the referenced range.

Reply:
@ivakegg You are correct that the QueryData object has the correct range for the referenced range; however, the SequentialScheduler runs ranges with a vanilla Accumulo Batch Scanner: one IteratorSetting per group of ranges. The provided QueryData object can have pruned queries, meaning that any given QueryData range may have an insufficient query for all ranges since RangeStream's process reduces the query.

One way to solve the problem was to take the query as it entered RangeStream (  expanded, flattened, and validated ), but there are other possible solutions, such as combining queries from all QueryData objects, and disabling RangeStream functions when a different scheduler is used. Do those seem more palatable?

Reply:
Fundamentally we're pitting the time savings of the sequential scheduler against the time savings of pruning the query tree in the range stream. Has either cost been quantified? At what point would it make sense to send a query down either path?

I suspect that for a subset of sufficiently simple queries the sequential scheduler will indeed be faster, but there still exists a large number of queries where pruning via range stream is still desirable. 

Combining all queries from all QueryData objects sounds attractive, but difficult to get right the first time.

Reply:
I have another PR that I'm prepping for RangeStream ( currently resolving a few unit tests) that may help on the performance front, but this PR is simply focusing on the bug fix, which is that we can miss results if the sequential scheduler is used. 

You are correct @apmoriarty , this was tested as a means of comparing and contrasting schedulers and trying to determine when and where either might be more beneficial. In that testing missing results were identified. The choice of using the pre-RangeStream query is out of simplicity since it already exists within the query planner. Tom added some tests as you can see, which I think is a huge help.

Reply:
Nope, let's get this in!